src/pybind/mgr/dashboard/services/auth.py

```diff
@@ -1,13 +1,101 @@
 # -*- coding: utf-8 -*-
 from __future__ import absolute_import
 
+from base64 import b64encode
+from datetime import datetime, timedelta
+import json
+import os
+import threading
 import time
+import uuid
 
 import cherrypy
+import jwt
 
-from .access_control import LocalAuthenticator
+from .access_control import LocalAuthenticator, UserDoesNotExist
 from .. import mgr, logger
-from ..tools import Session
+
+
+class JwtManager(object):
+    JWT_TOKEN_BLACKLIST_KEY = "jwt-token-black-list"
+    JWT_TOKEN_TTL = 28800  # default 8 hours
```

Comment:
why 8 hours?

Reply:
The idea is for the token to endure a full working day.